src/analysis/normed_space/lp_space.lean

```diff
@@ -629,6 +629,62 @@ end
 
 end normed_space
 
+section normed_star_group
+
+variables [Π (i : α), star_add_monoid (E i)] [Π i, normed_star_group (E i)]
+
+lemma _root_.mem_ℓp.star_mem {f : Π i, E i}
+(hf : mem_ℓp f p) : mem_ℓp (star f) p :=
+begin
+  rcases p.trichotomy with rfl | rfl | hp,
+  { apply mem_ℓp_zero,
+    simp [hf.finite_dsupport] },
+  { apply mem_ℓp_infty,
+    simpa using hf.bdd_above },
+  { apply mem_ℓp_gen,
+    simpa using hf.summable hp },
+end
+
+instance : has_star (lp E p) :=
+{ star := λ f, ⟨(star f : Π i, E i) , f.property.star_mem ⟩}
+
+@[simp] protected theorem star_apply (f : lp E p) (i : α) : star f i = star (f i) := rfl
+
+instance : has_involutive_star (lp E p) := {star_involutive := λ x,
+begin
+ext i,
+simp,
+end
+}
+
+instance : star_add_monoid (lp E p) := {star_add :=
+λ f g, by ext i ; simp only [star_add_monoid.star_add, lp.coe_fn_add, add_left_inj, pi.add_apply,
+  lp.star_apply, eq_self_iff_true]}
+
+ instance [hp : fact (1 ≤ p)] : normed_star_group (lp E p) := { norm_star :=
+ begin
+ intro f,
+ rcases p.trichotomy with h | h | h,
+ { unfreezingI { subst h }, exfalso,
+   have := ennreal.to_real_mono ennreal.zero_ne_top hp.elim,
+   norm_num at this,},
+ { unfreezingI { subst h }, simp only [lp.norm_eq_csupr, lp.star_apply, norm_star]},
+ { simp only [lp.norm_eq_tsum_rpow h, lp.star_apply, norm_star]}
```

Comment:
in every begin-end block we indent by one tab/two spaces.

Reply:
OK. Will do. Thanks!